scripts/profiler/stats_nordic.py

```diff
@@ -57,68 +53,48 @@ def calculate_stats_preset1(self, start_meas, end_meas):
+        event_type_id = self.processed_data.get_event_type_id(event_name)
+        if event_type_id is None:
             self.logger.error("Event name not found: " + event_name)
             return None
 
         trackings = list(filter(lambda x:
                       x.submit.type_id == event_type_id,
                       self.processed_data.tracked_events))
 
-        if type(event_state) is not EventState:
+        if not isinstance(event_state, EventState):
             self.logger.error("Event state should be EventState enum")
             return None
 
         if event_state == EventState.SUBMIT:
             timestamps = np.fromiter(map(lambda x: x.submit.timestamp, trackings),
                                      dtype=np.float)
         elif event_state == EventState.PROC_START:
-            timestamps = np.fromiter(map(lambda x: x.proc_start_time, trackings),
+            timestamps = np.fromiter(map(lambda x: x.submit.timestamp, trackings),
                                      dtype=np.float)
         elif event_state == EventState.PROC_END:
-            timestamps = np.fromiter(map(lambda x: x.proc_end_time, trackings),
+            timestamps = np.fromiter(map(lambda x: x.submit.timestamp, trackings),
                                      dtype=np.float)
 
         timestamps = timestamps[np.where((timestamps > start_meas)
                                          & (timestamps < end_meas))]
 
         return timestamps
 
-    def calculate_times_between(self, start_times, end_times):
+    @staticmethod
+    def calculate_times_between(start_times, end_times):
         if end_times[0] <= start_times[0]:
             end_times = end_times[1:]
         if len(start_times) > len(end_times):
             start_times = start_times[:-1]
 
         return (end_times - start_times) * 1000
 
-    def prepare_stats_txt(self, times_between):
-        stats_text = "Max time: "
-        stats_text += "{0:.3f}".format(max(times_between)) + "ms\n"
-        stats_text += "Min time: "
-        stats_text += "{0:.3f}".format(min(times_between)) + "ms\n"
-        stats_text += "Mean time: "
-        stats_text += "{0:.3f}".format(np.mean(times_between)) + "ms\n"
-        stats_text += "Std dev of time: "
-        stats_text += "{0:.3f}".format(np.std(times_between)) + "ms\n"
-        stats_text += "Median time: "
-        stats_text += "{0:.3f}".format(np.median(times_between)) + "ms\n"
-        stats_text += "Number of records: {}".format(len(times_between)) + "\n"
-
-        return stats_text
-
     def time_between_events(self, start_event_name, start_event_state,
                             end_event_name, end_event_state, hist_bin_width=0.01,
                             start_meas=0, end_meas=float('inf')):
         self.logger.info("Stats calculating: {}->{}".format(start_event_name,
                                                             end_event_name))
-        if not self.processed_data.tracking_execution:
```

Comment:
We still should verify if event processing is tracked (just do not use self.processed_data.tracking_execution)

Reply:
It's now done in function `_get_timestamps` in line 60